widgets/toggle.js

```diff
@@ -38,21 +56,29 @@ function toggle({
     }),
     render: function({helper, results}) {
       var isRefined = helper.hasRefinements(facetName);
+      var values = find(results.getFacetValues(facetName), {name: isRefined.toString()});
 
-      function toggleFilter() {
-        var methodToCall = isRefined ? 'removeFacetRefinement' : 'addFacetRefinement';
-        helper[methodToCall](facetName, true).search();
-      }
+      var facetValue = {
+        name: label,
+        isRefined: isRefined,
+        count: values && values.count || null
+      };
+
+      var _templates = {
+        header: templates.header,
+        item: templates.body,
+        footer: templates.footer
+      };
 
       React.render(
-        <Toggle
-          isRefined={isRefined}
-          label={label}
-          template={template}
-          transformData={transformData}
+        <RefinementList
+          facetValues={[facetValue]}
+          templates={_templates}
+          cssClasses={cssClasses}
+          transformData={prepareData(transformData)}
           hideWhenNoResults={hideWhenNoResults}
           hasResults={results.hits.length > 0}
-          toggleFilter={toggleFilter}
+          toggleRefinement={toggleRefinement.bind(null, helper, facetName, facetValue.isRefined)}
```

Comment:
In the current implementation, the if check is done when toggleRefinement is done, not when the function is created. IMHO it's just best practice than real need here, your call.